### PR TITLE
Make the genesis keypair only accessible in unittest; Use more constants

### PR DESCRIPTION
**Pull request by Geod24**
Review commit-by-commit.

### CODE REVIEW

File: source/agora/common/crypto/Key.d [hunk 1]
```diff
@@ -105,12 +105,18 @@ public struct PublicKey
...
 
-    /// Constructor accepting only 1 argument
+    /// Construct an instance from the binary representation
     private this (typeof(PublicKey.tupleof) args) pure nothrow @safe @nogc
     {
         this.tupleof = args;
     }
 
+    /// Ditto
+    public this (const ubyte[] args) pure nothrow @safe @nogc
```
Comment (AndrejMitrovic) [inline source/agora/common/crypto/Key.d:115]: Unittest?

Comment (Geod24) [reply source/agora/common/crypto/Key.d:115]: It's called in the next commit

Comment (AndrejMitrovic) [reply source/agora/common/crypto/Key.d:115]: Ah I got a taste of my own medicine just now.

**Comment by AndrejMitrovic** [comment]
LGTM.

**Comment by AndrejMitrovic** [comment]
The failure is in `unit.BlockStorage`, however the code doesn't have to use `getGenesisKeyPair`. All the code tests for is the storage of blocks, not their validity.

I'll submit a PR fixing it so this PR can go green.

**Comment by AndrejMitrovic** [comment]
Actually it doesn't have to be a separate PR.

File: tests/unit/BlockStorage.d [hunk 2]
```diff
@@ -52,14 +52,14 @@ private void main ()
...
         KeyPair.random, KeyPair.random, KeyPair.random, KeyPair.random
     ];
-    Block[] blocks;
+    const(Block)[] blocks;
     Hash[] block_hashes;
 
     blocks ~= getGenesisBlock();
     storage.saveBlock(blocks[$ - 1]);
 
     auto gen_key_pair = getGenesisKeyPair();
```
Comment (AndrejMitrovic) [inline tests/unit/BlockStorage.d]:
```suggestion
     auto gen_key_pair = KeyPair.random();
```

**Review by AndrejMitrovic — APPROVED** [review]

**Comment by Geod24** [comment]
Green

**Pull request merged**